src/toolkits/activity_classification/ac_data_iterator.cpp

```diff
@@ -132,152 +132,291 @@ variant_map_type _activity_classifier_prepare_data_impl(const gl_sframe &data,
                                                         const int &predictions_in_chunk,
                                                         const std::string &target,
                                                         bool verbose) {
+#ifndef NDEBUG
```

Comment:
Sorry, but given the instability in the Python-side test for the data iteration (the crash we saw in Travis, I think it’s safer not to touch the original _activity_classifier_prepare_data_impl.

We can apply the previous review feedback to the partially copied new version, since that’s the one that will live on after we remove the Python toolkit.